staging/src/k8s.io/client-go/util/workqueue/queue.go

```diff
@@ -111,25 +111,32 @@ func (s set) delete(item t) {
 }
 
 // Add marks item as needing processing.
-func (q *Type) Add(item interface{}) {
+func (q *Type) Add(items ...interface{}) {
 	q.cond.L.Lock()
 	defer q.cond.L.Unlock()
-	if q.shuttingDown {
-		return
-	}
-	if q.dirty.has(item) {
+	if q.shuttingDown || len(items) == 0 {
```

Comment:
I think it's better to move the judgement `len(items) == 0` to head of this function, to avoid unnecessary locks and unlocks if no items were input.